src/runtime_src/core/tools/common/ReportAie.cpp

```diff
@@ -502,13 +504,26 @@ ReportAie::getPropertyTree20202(const xrt_core::device * _pDevice,
   _pt.add_child("aie_metadata", populate_aie(_pDevice, "Aie_Metadata"));
 }
 
-void 
+void
 ReportAie::writeReport(const xrt_core::device* /*_pDevice*/,
-                       const boost::property_tree::ptree& _pt, 
-                       const std::vector<std::string>& /*_elementsFilter*/, 
+                       const boost::property_tree::ptree& _pt,
+                       const std::vector<std::string>& _elementsFilter,
                        std::ostream & _output) const
 {
   boost::property_tree::ptree empty_ptree;
+  std::vector<std::string> aieCoreList;
+  bool is_less = false;
+
+  for (auto it = _elementsFilter.begin(); it != _elementsFilter.end(); ++it) {
+    if(*it == "cores") {
+      auto core_list = std::next(it);
+      if (core_list != _elementsFilter.end())

```

Comment:
Please do some negative testing around these options.

Reply:
I have done that. With some value provided or without
 provided any  value